src/com/visualization/piechart/piechart.js

```diff
@@ -0,0 +1,69 @@
+import {h, Component} 				from 'preact/preact';
+
+import PieSegment					from 'com/visualization/piechart/piesegment';
+
+export default class PieChart extends Component {
+
+	constructor( props ) {
+		super(props);
+
+    }
+
+    // Converts an array of number to an array of percentages
+    convertToPercentage ( values ) {
+        // calculate total
+        let total = 0;
+        values.forEach( ( v ) => ( total += v ) );
+
+        // scale values so there sum is 100%
+        let percentages = values.map( ( v ) => ( (100 * v) / total ) );
+
+        return percentages;
+    }
+
+    render( props ) {
+
+        if ( !(props && props['lables'] && props['values'])) {
+            console.warn('PieChart was created with invalid props', props);
+            return <div> No Data! </div>;
+        }
+
+        let lables = props.lables;
+        let values = props.values;
+
+        let percentages = this.convertToPercentage(values);
+
+        let Segments = [];
+        let Ledgend = [];
+
+        let offset = 0;
+        for (var i = 0; i < percentages.length; i++) {
```

Comment:
Prefer
```
for ( var i = 0; i < percentages.length; i++ ) {
```

Reply:
I'd prefer
```
percentages.forEach( (percentage, i) => { ...}
```

Reply:
classic for loops are a construct found in both JavaScript and PHP. 

Also: https://jsperf.com/for-vs-foreach/37

Reply:
Keeping it as the classic for, it's the most common loop type when you need the index.